File: gym-backend/src/main/java/com/in/GymManagementSystem/controller/MemberService.js
import axios from "axios";

const API_URL = "http://localhost:9999/api/members";

export const getAllMembers = async () => {
  const response = await axios.get(API_URL);
  return response.data;
};

export const getMemberById = async (id) => {
  const response = await axios.get(`${API_URL}/${id}`);
  return response.data;
};

export const createMember = async (memberData) => {
  const response = await axios.post(API_URL, memberData);
  return response.data;
};

export const updateMember = async (id, memberData) => {
  const response = await axios.put(`${API_URL}/${id}`, memberData);
  return response.data;
};

export const deleteMember = async (id) => {
  await axios.delete(`${API_URL}/${id}`);
};

export const assignTrainer = async (memberId, trainerId) => {
  const response = await axios.post(`${API_URL}/${memberId}/trainers/${trainerId}`);
  return response.data;
};

export const removeTrainer = async (memberId, trainerId) => {
  const response = await axios.delete(`${API_URL}/${memberId}/trainers/${trainerId}`);
  return response.data;
};

File: gym-backend/src/main/java/com/in/GymManagementSystem/service/MemberService.java
package com.in.GymManagementSystem.service;

import com.in.GymManagementSystem.dto.MemberDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface MemberService {
    List<MemberDTO> getAllMembers();

    Page<MemberDTO> getAllMembersPaged(Pageable pageable);

    MemberDTO getMemberById(Long id);

    MemberDTO createMember(MemberDTO memberDTO);

    MemberDTO updateMember(Long id, MemberDTO memberDTO);

    void deleteMember(Long id);

    MemberDTO assignTrainer(Long memberId, Long trainerId);

    MemberDTO removeTrainer(Long memberId, Long trainerId);
}
